add nix package and nixosModule

e-v-o-l-v-e opened the pull request:
Hello, if you're open to this, it just adds a flake.nix and a flake.lock to allow use with Nix.

It will not require anything more from you, and it's mostly temporary, I already submitted a PR to add the package it in nixpkgs (the repo for nix packages) and I plan to make it a nixos module afterwards.

I'll update the version number and the flake.lock when needed.

thank you for the software btw, it's really cool, I use it all the time.

Tanq16 commented:
hey @e-v-o-l-v-e , thanks for this PR - glad you like the app

i've not played with nix and am running low on time, so give me a bit to review, will merge this soon

e-v-o-l-v-e commented:
no problems ! if you're interested i recommend https://nixos-and-flakes.thiscute.world/

I've just refactored the flake to allow use on different architecture, and rebased it to have only one commit.

No problem take your time, in the meantime if someone wants to try it they can do so with [my fork](https://github.com/e-v-o-l-v-e/local-content-share) : 
```sh
nix run github:e-v-o-l-v-e/local-content-share
```

e-v-o-l-v-e committed:
add flake (output package and nixosModule)

e-v-o-l-v-e commented:
I basically rewrote everything but better, now support any unix system whereas it was only x86 linux, and added the nixos module because for some reason it is still not merged in nixpkgs.

the readme update is the important part though, i added everything as 3 folding section so it doesn't take too much space, feel free to ask me any question, I can add comments if what the flake does isn't clear

the link update is really cool, still an amazing software, keep up the good work ^^

(if someone wants to use the nixosModule before this gets merge you just need to use my fork as the inputs :

```nix
    local-content-share = {
      url = "github:e-v-o-l-v-e/local-content-share";
      inputs.nixpkgs.follows = "nixpkgs";
    };
```

e-v-o-l-v-e commented:
hey, sorry for the delay, I just got back to school so not a lot of free time :sweat_smile:, here are the changes, i kept the notive for the unmerged PR though, because it has been sitting there for 2 month and i don't know if it's ever gonna get merged :smiling_face_with_tear: 
I have tried setting up a GHA and bafflingly failed, I'll try again just now but will probably fail again
